acorn/src/expression.js

```diff
@@ -782,6 +808,10 @@ pp.parseProperty = function(isPattern, refDestructuringErrors) {
   return this.finishNode(prop, "Property")
 }
 
+pp.getGetterSetterExpectedParamCount = function(prop) {
```

Comment:
Also here -- how is this different in TypeScript?

Reply:
This situation will be supported in typescript 3.8.3. [playground link](https://www.typescriptlang.org/play?ts=3.8.3#code/MYewdgzgLgBNCuATApmWBeGBvAUDfMAZgJYBO0AcgIYC2yAXDAORTLRMA0eB3+A9HyrBgbCCFIwADqRCTkpKAE8AFBGRRWpAJS846mMAAWVMAHNk1OsqiHiEDjDDIA7pYZwopYma3ZdBGF0AXxwggG4gA)
```
const student = {
    firstName: 'test',
    
    //accessor property(setter)
    set changeName(this, newName: string) {
        
    }
};
```

Reply:
Right. But that still is a *very* specific method. And even with your changes it seems `set x(this, ...foo)` would still get through. Would it work to have `parseGetter`/`parseSetter` methods that are called from `parseProperty`, so you have less code to duplicate, but we don't lock ourselves into this kind of mini-method-protocol?

Reply:
Its a great idea to add `parseGetter`/`parseSetter` methods. And I think it will be ez to extend other logic in the future.